packages/auth/src/AuthService.ts

```diff
@@ -0,0 +1,63 @@
+import Keycloak from "keycloak-js";
+import {KeycloakError, KeycloakInitOptions, KeycloakInstance, KeycloakProfile, KeycloakPromise} from "keycloak-js";
+import { Key } from "readline";
+
+/**
+ * Wrapper class for {Keycloak.KeycloakInstance}
+ */
+export class AuthService {
+
+    private auth: KeycloakInstance;
+
+    constructor(config: KeycloakInitOptions) {
+      this.auth = Keycloak(config);
+    }
+
+  /**
+   * Called to initialize the adapter.
+   * @param initOptions Initialization options.
+   * @returns A promise to set functions to be invoked on success or error.
+   */
+  public init(initOptions: KeycloakInitOptions): KeycloakPromise<boolean, KeycloakError> {
+      if (!initOptions.onLoad) {
+        initOptions.onLoad = "check-sso";
+      }
+      return this.auth.init(initOptions);
+    }
+
+    /**
+		 * Loads the user's profile.
+		 * @returns A promise to set functions to be invoked on success or error.
+		 */
+    public loadUserProfile(): KeycloakPromise<KeycloakProfile, void> {
```

Comment:
-1 for exposing a custom promise type, is it needed? does it have any extra useful methods? Maybe just return a `Promise` if it's castable.